agent/session_persister.py

```
@@ -0,0 +1,478 @@
+        if self._session_db:
+            try:
+                self._session_db.end_session(old_session_id, "compression")
+                self.session_id = new_id  # Atomic update via setter
+                self._flushed_msg_count = 0  # Reset for new session
+                self._session_db.create_session(
+                    session_id=self._session_id,
+                    source=platform or self._platform or "cli",
+                    model=model or self._model,
+                    parent_session_id=parent_session_id or old_session_id,
+                )
+            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)